Lib/pathlib.py

```diff
@@ -873,6 +873,10 @@ def absolute(self):
             cwd = self._flavour.abspath(self.drive)
         else:
             cwd = os.getcwd()
+            if not self.root and not self._tail:
```

Comment:
This is very clever! Could we add a comment so it's immediately obvious what's going on here:

```suggestion
            # fast path for "empty" paths, e.g. Path("."), Path() or Path("")
            # Path() is significant, as it's called from the cwd() class method
            if not self.root and not self._tail:
```

Reply:
The suggestion isn't quite right --  we don't call the `Path.cwd()` classmethod from `absolute()`. The performance improvement comes from two places:

1. We supply _one_ argument to `with_segments()` rather than two. As a result, `PurePath.__init__()` does not need to call `os.path.join()`.
2. We store the result of `getcwd()` on `path._str` because it's guaranteed by Posix to be normalized (e.g. no doubled slashes or `./` segments). The same appears to be true on Windows (based on googling and ad hoc testing) though I couldn't find it specified anywhere.

Reply:
(I'll add a comment to clarify)

Reply:
I meant that we call `.absolute()` from `.cwd()`, not the other way around!

Reply:
Oh I see! I'll edit the comment

Reply:
> 2\. We store the result of `getcwd()` on `path._str` because it's guaranteed by Posix to be normalized (e.g. no doubled slashes or `./` segments). The same appears to be true on Windows (based on googling and ad hoc testing) though I couldn't find it specified anywhere.

On Windows, the current directory is stored as a normalized path in the PEB `ProcessParameters`. Unlike POSIX `getcwd()`, WinAPI `GetCurrentDirectoryW()` doesn't return a real path, and Python's implementation of `os.getcwd()` on Windows doesn't implement this POSIX requirement. One can call WinAPI `CreateFileW()` and `GetFinalPathNameByHandleW()` (i.e. `nt._getfinalpathname()`) to get the final path, if working with a real path matters. For example:

```text
0:001> ?? @$peb->ProcessParameters->CurrentDirectory
struct _CURDIR
   +0x000 DosPath          : _UNICODE_STRING "C:\Temp\test\spamlink\"
   +0x010 Handle           : 0x00000000`00000050 Void
```

```pycon
>>> os.getcwd()
'C:\\Temp\\test\\spamlink'
>>> nt._getfinalpathname('.')
'\\\\?\\C:\\Temp\\test\\spam'
```

Reply:
That's mega, thanks Eryk. For the purpose of this PR it doesn't matter that the path isn't guaranteed to be real, only that it's guaranteed normalized, so I think we're in the clear!